cluster-autoscaler/cloudprovider/azure/azure_cache.go

```diff
@@ -29,7 +29,10 @@ import (
 	"k8s.io/klog"
 )
 
-var virtualMachineRE = regexp.MustCompile(`^azure://(?:.*)/providers/Microsoft.Compute/virtualMachines/(.+)$`)
+var (
+	defaultAsgCacheTTL = 900 // Seconds
```

Comment:
A comment somewhere as to how/why this prevents the throttling issue would be appreciated by future maintainers.

What is the rationale in changing this from 3600 seconds (1 hour) to 900 seconds (15 minutes)?  I am not against it other than I would like to know why.

Reply:
smaller TTL actually increases the chances of throttling. I think we can change the default TTL back to 1 hour, and still keep it tunable